lib/PgSQL_Monitor.cpp

```
@@ -117,7 +124,179 @@ void check_and_build_standard_tables(SQLite3DB& db, const vector<table_def_t>& t
+				}
+			}
+			return pg;
+		}
+	}
+	return pg;
+}
+
+void PgSQL_Monitor_Connection_Pool::put_connection(char* hostname, int port, PGconn* pg) {
+	std::lock_guard<std::mutex> lock(mutex);
+	MonPgConn_t* mc = (MonPgConn_t*)malloc(sizeof(MonPgConn_t));
+	mc->conn = pg;
```

Cppcheck (2.21.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)